### PR TITLE
Fix edge case for new apartment sell by dates

### PR DESCRIPTION
**Pull request by matti-lamppu**
# Hitas Pull Request

# Description

Fixes `sell_by_date` calculation in cases where the apartment is completed before it is sold for the first time. 
In these cases, the `sell_by_date` should be calculated based on the first sale purchase date.

A bit of refactoring is also made to reuse code and avoid import loops.

## Pull request checklist

Check the boxes for each DoD items that has been completed:

- **Frontend**
    - [ ] Changes have been tested
- **Backend**
    - [x] Changes have been tested
    - [x] Automatic tests has been added
    - [ ] OpenAPI definitions have been updated
    - [ ] Database migrations will work in test environment
    - [ ] Oracle migration has been updated
    - [ ] initial.json has been updated to work with migrations

## Test plan

- [x] Automatic tests

## Tickets

This pull request resolves all or part of the following ticket(s): HT-459

### CODE REVIEW

File: backend/hitas/services/condition_of_sale.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+from django.db import models
+from django.db.models import Q
+from django.utils import timezone
+
+from hitas.models import ConditionOfSale, Owner, Ownership
+from hitas.models.condition_of_sale import ConditionOfSaleAnnotated
+from hitas.services.apartment import subquery_first_purchase_date
+
+
+def condition_of_sale_queryset() -> models.QuerySet[ConditionOfSaleAnnotated]:
```
Comment (ranta) [inline backend/hitas/services/condition_of_sale.py:11]: Refactoring code should be in their own commits, not mixed in the same commit with unrelated changes.

**Review by ranta — APPROVED** [review]

File: backend/hitas/services/condition_of_sale.py (hunk 1 above)
Comment (matti-lamppu) [reply backend/hitas/services/condition_of_sale.py:11]: I refactored these because I initially had the apartment and condition of sale functions in the respective model's files, but that caused import loops, so I had to move both of them to a new place.
